### PR TITLE
feat(plugin-server): Manage webhook SITE_URL logic via instance settings

### PR DESCRIPTION
**Pull request by macobo**
## Problem

See https://github.com/PostHog/posthog/issues/9790 for context.

## Changes

This PR makes us manage plugin-server side SITE_URL logic via instance settings.

Previously, we parsed the site url from an incoming request and passed it to webhooks for formatted messages. Now we save that same url to INGESTION_SITE_URL var and use that (or the previously-set SITE_URL var) to determine how to link users.

Also implemented a follow-up for a previous PR - combined as they touch the same file to avoid yet more merge conflicts.

## How did you test this code?

See tests.

### CODE REVIEW

File: plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts [hunk 1]
```diff
@@ -10,25 +10,40 @@ export async function runAsyncHandlersStep(
...
+    })
+}
+
+async function processOnActionAndWebhooks(
+    runner: EventPipelineRunner,
+    event: PreIngestionEvent,
+    person: Person | undefined,
+    elements: Element[] | undefined
+) {
     const promises = []
```
Comment (yakkomajuri) [inline plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts:38]: this is no longer used for anything

File: plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts [hunk 2]
```diff
@@ -10,25 +10,40 @@ export async function runAsyncHandlersStep(
...
     let actionMatches: Action[] = []
+    const processedPluginEvent = convertToProcessedPluginEvent(event)
+
     if (event.event !== '$snapshot') {
         actionMatches = await runner.hub.actionMatcher.match(event, person, elements)
-        promises.push(runner.hub.hookCannon.findAndFireHooks(event, person, event.siteUrl, actionMatches))
+        promises.push(runner.hub.hookCannon.findAndFireHooks(event, person, actionMatches))
```
Comment (yakkomajuri) [inline plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts:44]: note that we're no longer awaiting these anywhere - we don't return them so `runAsyncHandlers` step doesn't await them. This function will just fire these promises and forget about them.

File: posthog/settings/dynamic_settings.py [hunk 3]
```diff
@@ -108,6 +108,7 @@
         "Used to disable emails from async migrations service",
         bool,
     ),
+    "INGESTION_SITE_URL": (None, "Used in ingestion pipeline to determine sites url", str),
```
Comment (yakkomajuri) [inline posthog/settings/dynamic_settings.py:111]: do we really need this? I can see how messing with `SITE_URL` might be annoying but I feel this is confusing to users - they will be very confused as to why there's `SITE_URL` and `INGESTION_SITE_URL` and they're both set differently.

Comment (macobo) [reply posthog/settings/dynamic_settings.py:111]: Note this is not shown in instance settings. :)

File: plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts (hunk 2 above)
Comment (macobo) [reply plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts:44]: This is a bug, good catch. There should be a Promise.all down here :)

Will fix.

File: posthog/settings/dynamic_settings.py (hunk 3 above)
Comment (yakkomajuri) [reply posthog/settings/dynamic_settings.py:111]: True - got confused quickly reading here. Point still stands but less relevant

Comment (macobo) [reply posthog/settings/dynamic_settings.py:111]: Not sure what that point is. INGESTION_SITE_URL is purely managed by plugin-server and is a setting we're using to do this whole dance.

**Review by yakkomajuri — APPROVED** [review]

**Pull request merged**